packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1725,6 +1726,33 @@ function validateAttributes(
     }
 }
 
+function validateSlotAttribute(
+    ctx: ParserCtx,
+    parsedAttr: ParsedAttribute,
+    parentNode: ParentNode,
+    element: BaseElement
+): void {
+    const slotAttr = parsedAttr.get('slot');
+
+    if (!slotAttr) {
+        return;
+    }
+
+    if (
+        ast.isElement(parentNode) ||
+        ast.isSlot(parentNode) ||
+        ast.isForEach(parentNode) ||
+        ast.isForOf(parentNode)
```

Comment:
Actually for-loops work fine. This slots without a problem:

```html
    <x-counter>
        <template for:each={items} for:item="item">
            <div slot="foo" key={item}>
                Slot me into "foo" please
            </div>
        </template>
    </x-counter>
```

This works in [native shadow](https://stackblitz.com/edit/salesforce-lwc-qx3vga?file=src%2Fmodules%2Fx%2Fapp%2Fapp.html&title=LWC%20playground), [synthetic shadow](https://stackblitz.com/edit/salesforce-lwc-wbqqg2?file=src%2Fmain.js&title=LWC%20playground), and [light DOM](https://stackblitz.com/edit/salesforce-lwc-ie1zxm?file=src%2Fmodules%2Fx%2Fcounter%2Fcounter.html&title=LWC%20playground).